### PR TITLE
K8s

### PR DESCRIPTION
**Pull request by Naramsim**
Hello everyone,

I had to pick up k8s again for a certification and thought about creating a deployment for PokeAPI. I don't know if anyone that contributes to this repository is familiar with k8s, but anyway, feel free to ask as many questions as you want.

I used Kustomize to allow anyone to deploy a cluster with their own secrets/config in the simplest way. Basically one needs to configure `kubectl` to point to a cluster and then they have to run `kubectl -k apply` and voila' a copy of pokeapi will be deployed on the cluster. In the [`README.md`](https://github.com/PokeAPI/pokeapi/tree/k8s#kubernetes--) I wrote I small guide about how to set everything up, you can read that to see how to build the data etc.

The k8s files are tested as well with a Github action.

--- 

This PR also builds and pushes different platform Docker images. Namely: linux/amd64, linux/arm64, linux/ppc64le, linux/arm/v7, linux/arm/v6

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -14,7 +14,6 @@ db.*
 venv*
 node_modules
 Resources
-graphql
```
Comment (phalt) [inline .dockerignore]: I remember you added this in another PR, might need to rebase vs master.

File: .github/workflows/kustomize.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      - name: Migrate and build data
+        run: |
+          make k8s-migrate
+          make k8s-build-db
+          bash Resources/scripts/wait.sh http://localhost:8001/api/v1/namespaces/pokeapi/services/pokeapi/proxy/api/v2/pal-park-area/5/
+      - name: K8s Apply
+        run: |
+          kubectl wait --timeout=120s --for=condition=complete job/load-graphql
+          last_command=$(kubectl get job -o jsonpath='{.status.succeeded}' load-graphql)
+          test "$last_command" -eq 1
```
Comment (phalt) [inline .github/workflows/kustomize.yml]: newline at end of file?

**Review by phalt — CHANGES_REQUESTED** [review]

**Comment by Naramsim** [comment]
I've added a yamllint job to check the syntax

**Comment by Naramsim** [comment]
Forcing merge

**Comment by pokeapi-machine-user** [comment]
A [PokeAPI/api-data](https://github.com/PokeAPI/api-data) refresh has started. In 45 minutes the staging branch of [PokeAPI/api-data](https://github.com/PokeAPI/api-data/tree/staging) will be pushed with the new generated data. <br><br> The staging branch will be deployed in our [staging environment](https://staging.pokeapi.co/api/v2/) and you will be able to review the entire API. <br><br> A Pull Request ([master](https://github.com/PokeAPI/api-data/tree/master)<-[staging](https://github.com/PokeAPI/api-data/tree/staging)) will be also created at [PokeAPI/api-data](https://github.com/PokeAPI/api-data/pulls) and assigned to the PokeAPI Core team to be reviewed. If approved and merged new data will soon be available worldwide at [pokeapi.co](https://pokeapi.co/api/v2/).

**Pull request merged**